Source/com/drew/imaging/png/PngMetadataReader.java

```diff
@@ -178,30 +183,69 @@ private static void processChunk(@NotNull Metadata metadata, @NotNull PngChunk c
             metadata.addDirectory(directory);
         } else if (chunkType.equals(PngChunkType.tEXt)) {
             SequentialReader reader = new SequentialByteArrayReader(bytes);
-            byte[] keywordBytes = reader.getNullTerminatedBytes(79);
-            StringValue keyword = new StringValue(keywordBytes, _latin1Encoding);
-            int bytesLeft = bytes.length - keywordBytes.length - 1;
+            String keyword = reader.getNullTerminatedStringValue(79, _latin1Encoding).toString();
+            int bytesLeft = bytes.length - keyword.length() - 1;
             StringValue value = reader.getNullTerminatedStringValue(bytesLeft, _latin1Encoding);
             List<KeyValuePair> textPairs = new ArrayList<KeyValuePair>();
             textPairs.add(new KeyValuePair(keyword, value));
             PngDirectory directory = new PngDirectory(PngChunkType.iTXt);
             directory.setObject(PngDirectory.TAG_TEXTUAL_DATA, textPairs);
             metadata.addDirectory(directory);
+        } else if (chunkType.equals(PngChunkType.zTXt)) {
+            SequentialReader reader = new SequentialByteArrayReader(bytes);
+            String keyword = reader.getNullTerminatedStringValue(79, _latin1Encoding).toString();
+            byte compressionMethod = reader.getInt8();
+
+            int bytesLeft = bytes.length - keyword.length() - 1 - 1 - 1 - 1;
+            byte[] textBytes = null;
+            if (compressionMethod == 0) {
+                try {
+                    // Creates a new decompressor. If the parameter 'nowrap' is true then the ZLIB header and checksum fields will not be used.
+                    // This provides compatibility with the compression format used by both GZIP and PKZIP.
+                    Inflater inf = new Inflater(true);
+                    textBytes = StreamUtil.readAllBytes(new InflaterInputStream(new ByteArrayInputStream(bytes, bytes.length - bytesLeft, bytesLeft), inf));
+                } catch(java.util.zip.ZipException zex) {
+                    textBytes = null;
+                    // TODO: should zlib/deflate errors be returned?
+                    //PngDirectory directory = new PngDirectory(PngChunkType.zTXt);
+                    //directory.addError("PNG zTXt error: " + zex.getMessage());
+                    //metadata.addDirectory(directory);
```

Comment:
Sorry - meant to get back with you. Part of this concern was .NET couldn't catch the exception with yield, but your last commits there fixes that. I'm happy to change this as well unless you already started.